tools/NetCoreConversion/NetCoreConversion/Utility/StringExtensions.cs

```diff
@@ -0,0 +1,275 @@
+            var privateDataFollower = privateData == null ? String.Empty : $" {CommentPrefix}End of {privateDataDisplayName} hashtable";
+            foreach (var privateDataLine in privateData.ToDefinitionEntry().ToList().UpdateFirstAndLast(privateDataLeader, privateDataFollower))
+            {
+                yield return privateDataLine;
+            }
+
+            yield return String.Empty;
+        }
+
+        public static IEnumerable<string> ToDefinitionEntry<TClass, TList>(this TClass @class, Expression<Func<TClass, List<TList>>> propertySelector)
+        {
+            if (GetDescription(propertySelector, out var description)) yield return description;
+            (var values, var firstLeader) = GetValue(@class, propertySelector);
+
+            var isStringList = typeof(TList) == typeof(string);
+            var listPrefix = isStringList ? String.Empty : ObjectListPrefix;
+            var listPostfix = isStringList ? String.Empty : ObjectListPostfix;
+            var valuePrefix = isStringList ? ElementPrefix : String.Empty;
+            var valuePostfix = isStringList ? ElementPostfix : String.Empty;
+
+            foreach (var propertyValue in CreateListValues(values, firstLeader, listPrefix, listPostfix, valuePrefix, valuePostfix, isStringList))
+            {
+                yield return propertyValue;
+            }
+
+            yield return String.Empty;
+        }
+
+        private const int MaxLineLength = 70;
+        private static IEnumerable<string> CreateListValues<TList>(IReadOnlyList<TList> values, string firstLeader, 
+            string listPrefix, string listPostfix, string valuePrefix, string valuePostfix, bool hasMaxLengthRestriction)
+        {
+            if (values == null || !values.Any())
+            {
+                yield return $"{firstLeader}{ObjectListPrefix}{ObjectListPostfix}";
+                yield break;
+            }
+
+            var currentLine = String.Empty;
+            var firstLine = true;
+
+            string GetLeader() => firstLine ? $"{firstLeader}{listPrefix}" : Indent;
+            bool IsFirst(int index) => index == 0;
+            bool IsLast(int index) => index == values.Count - 1;
+
+            for (var i = 0; i < values.Count; ++i)
+            {
+                var line = $"{valuePrefix}{values[i]}{valuePostfix}{(IsLast(i) ? listPostfix : ", ")}";
+                var priorCurrentLine = currentLine;
+                currentLine += line;
+                if (IsFirst(i) || hasMaxLengthRestriction && currentLine.Length <= MaxLineLength) continue;
+
+                yield return $"{GetLeader()}{priorCurrentLine}";
+                currentLine = line;
+                firstLine = false;
+            }
+            if (currentLine != String.Empty)
+            {
+                yield return $"{GetLeader()}{currentLine}";
+            }
+        }
+
+        public static IEnumerable<string> ToDefinitionEntry(this PsDefinitionHeader header)
```

Comment:
Should not mix public extension methods with other public static methods - please add these to an extensions class.

Reply:
This is a public extension method that turns a `PsDefinitionHeader` into a collection of strings. It makes sense to make all these extensions (`ToDefinitionEntry`) in the same place since they all serve the same purpose/same context.

Reply:
I missed that the other static methods that weren't extensions were private, so this is just fine.